api/app/controllers/webhook.controller.js

```diff
@@ -0,0 +1,48 @@
+const db = require("../sequelize/models");
+const User = db.user;
+const Review = db.review;
+const Transcript = db.transcript;
+const Op = db.Sequelize.Op;
+
+
+exports.create = async (req, res) => {
+    // const action = req.headers["x-github-event"];
+    const pull_request = req.body;
+     const action = pull_request.action;
+
+
+    // Check if it's a pull request event
+    if (action === 'opened' || action === 'closed' || action === 'synchronize') {
+        const html_url = pull_request.pull_request.html_url;
+
+        console.log(`Received pull request event for PR: ${html_url}`);
+
+        try {
+            // Check if the PR URL exists in the database
+            const existingReview = await Transcript.findOne({ where: { pr_url: html_url } });
+
+            if (existingReview) {
+                // PR is merged, update the mergeAt timestamp
+                existingReview.mergedAt = new Date();
+                existingReview.archivedAt = new Date();
+
+                await existingReview.save();
+
```

Comment:
And lastly, requeue the transcript if the PR wasn't merged.
What do you think?

Reply:
requeing is a function of time yea?

Reply:
what is the condition for requeue?

Reply:
@adamjonas any comment on this??

Reply:
if a PR is opened within the expiration period but no action is taken on the PR, the transcript is not requeued until the PR is either merged or archived (closed) - this is reflected by time stamps on the respective fields on the review table 

if no PR is opened after the review expires, the transcript is requeued (and a v2 feature is to use either the edited version OR the original when next claimed)